### PR TITLE
Removed manylinux2010 python artifacts

### PR DESCRIPTION
**Pull request by veblush**
Removing `manylinux2010` and leaving `manylinux2014` because
-  `manylinux2010` hit EOL in 2020 and it's going to be removed this year by https://github.com/pypa/manylinux/issues/1281. 
-  From the pypi stat, 3.8% of downloads for gcsio 1.46.1 was manylinux2010. (91.6% was manylinux2014). It's mostly because they're using old pip.
- `manylinux2010` image has a weird setup of gcc and glibc; gcc supports C11 but glibc doesn't have C11 headers, resulting in undefined symbols errors when building a recent boringssl. It tries to use C11 features because gcc claims it supports C11 but corresponding C11 headers/symbols are not available.

### CODE REVIEW

**Review by lidizheng — APPROVED** [review]
Thanks for working on this.

File: tools/run_tests/artifacts/artifact_targets.py [hunk 1]
```diff
@@ -442,16 +442,6 @@ def targets():
...
-        PythonArtifact('manylinux2010', 'x64', 'cp36-cp36m'),
-        PythonArtifact('manylinux2010', 'x64', 'cp37-cp37m', presubmit=True),
-        PythonArtifact('manylinux2010', 'x64', 'cp38-cp38'),
-        PythonArtifact('manylinux2010', 'x64', 'cp39-cp39'),
-        PythonArtifact('manylinux2010', 'x86', 'cp36-cp36m'),
-        PythonArtifact('manylinux2010', 'x86', 'cp37-cp37m', presubmit=True),
-        PythonArtifact('manylinux2010', 'x86', 'cp38-cp38'),
-        PythonArtifact('manylinux2010', 'x86', 'cp39-cp39'),
-        PythonArtifact('manylinux2014', 'aarch64', 'cp36-cp36m',
-                       presubmit=True),
```
[the remark this answers is not in the document]
Comment (veblush) [reply tools/run_tests/artifacts/artifact_targets.py:454]: Oops! Good catch! Fixed.

**Comment by veblush** [comment]
[grpc_build_artifacts_multiplatform](https://fusion2.corp.google.com/invocations/cc93151a-a337-47e6-92ed-11aa650e6c2d): running

**Review by jtattermusch — APPROVED** [review]
LGTM.

**Pull request merged**